packages/ui-list/src/InlineList/__new-tests__/InlineList.test.tsx

```diff
@@ -56,10 +56,8 @@ describe('<InlineList />', () => {
         <InlineList.Item>List item 2</InlineList.Item>
         <InlineList.Item>List item 3</InlineList.Item>
         <InlineList.Item>List item 4</InlineList.Item>
-        {/* eslint-disable-next-line no-constant-binary-expression */}
-        {null && <InlineList.Item>ignore me 1</InlineList.Item>}
-        {/* eslint-disable-next-line no-constant-binary-expression */}
-        {false && <InlineList.Item>ignore me 2</InlineList.Item>}
+        {null}
+        {false}
```

Comment:
new TS version is clever and figures out that the part after `&&` is never executed